### PR TITLE
Select last commit or full branch history for analysis of #bumps

### PR DESCRIPTION
**Pull request by sbe-arg**
Variable BRANCH_HISTORY" `full|last` allows to select the length of the history for finding bump keywords.

As per examples:
```
history='full'
declare -A history_type=( 
    ["last"]="$(git show -s --format=%B)" \
    ["full"]="$(git log master..HEAD --format=%B)" \
)
log=${history_type[${history}]}
printf "History:\n---\n%s\n---\n" "$log"
History:
---
history type


POC log
---
```

```
history='last'
declare -A history_type=( 
    ["last"]="$(git show -s --format=%B)" \
    ["full"]="$(git log master..HEAD --format=%B)" \
)
log=${history_type[${history}]}
printf "History:\n---\n%s\n---\n" "$log"
History:
---
history type
---
```

fixes https://github.com/anothrNick/github-tag-action/issues/211

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,7 +49,7 @@ jobs:
     steps:
     - uses: actions/checkout@v3
       with:
-        ref: ${{ github.event.pull_request.head.sha }}
+        ref: ${{ github.sha }}
```
Comment (sammcj) [inline README.md:52]: Do we even need this? By default it checks out the current ref (branch)

**Comment by sammcj** [comment]
Just that one tiny comment / question otherwise ok to approve!

File: README.md (hunk 1 above)
Comment (sbe-arg) [reply README.md:52]: Good point as is just a readme example I'm happy with either leaving it or removing the ref: altogether. up to you XD

**Review by sammcj — APPROVED** [review]
lgtm!

**Comment by sbe-arg** [comment]
Been sick and afk will merge in the next few days.

**Pull request merged**